swagger-spec/files/definition.yaml

```diff
@@ -0,0 +1,148 @@
+type: object
+title: File
+properties:
+  id:
+    type: string
+    readOnly: true
+    description: 'The identifier of the file entity.'
+  type:
+    type: string  
+    readOnly: true
+    description: 'The type identifier of the file entity (`files`).'
+  attributes:
+    type: object
+    readOnly: false
+    description:  'The properties of the file entity.'
+    properties:
+      extra:
+        type: object
+        readOnly: true
+        description: 'Extra information, contains `hashes` (`sha256`, `md5`) and `downloads` count.'
+      kind:
+        type: string
+        readOnly: true
+        description: 'The kind of files object it is (`file` or `folder`).'
+      name:
+        type: string
+        readOnly: false 
+        description: 'Name of the file'
+      last_touched:
+        type: string
+        format: date-time
+        readOnly: true
+        description: 'The time at which the file external metadata was last retrieved as an iso8601 formatted timestamp (does not apply to OSF Storage files).'
+      materialized_path:
+        type: string
+        readOnly: true
+        description: 'Unix-style path to the file relative to the provider root.'
+      date_modified:
+        type: string
+        format: date-time
+        readOnly: true
+        description: 'The time at which the file was last modified, as an iso8601 formatted timestamp.'
+      current_version:
+        type: integer
+        readOnly: true
+        description: 'Version number of the file.'
+      delete_allowed:
+        type: boolean
+        readOnly: true
+        description: 'Whether or not deletion of the file is allowed.'
+      date_created:
+        type: string
+        format: date-time
+        readOnly: true
+        description: 'The time at which the file was created, as an iso8601 formatted timestamp.'
+      provider:
+        type: string
+        readOnly: true
+        description: 'The id of the file provider (e.g., `osfstorage`)'
+      path:
+        type: string
+        readOnly: true
+        description: 'The unique identifier for this file entity for this project and storage provider.'
+      current_user_can_comment:
+        type: boolean
+        readOnly: true
+        description: 'Whether or not the current user is allowed to post comments.'
+      guid:
+        type: string
+        readOnly: true
+        description: 'Global unique identifier (GUID) for this file (if one has been assigned).'
+      checkout:
+        type: string
+        readOnly: true
+        description: 'SOON TO BE DEPRECATED, see relationships checkout.'
+      tags:
+        type: string
+        readOnly: true
+        description: 'An array of tags on the file (strings).'
+      size:
+        type: integer
+        readOnly: true
+        description: 'Size of the file in bytes.'
+  relationships:
+    type: object
+    readOnly: true
+    description: ''
```

Comment:
'URLs to other entities or entity collections that have a relationship to the file entity.'